torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -30,17 +30,44 @@ def _orthogonalize(matrix, epsilon=1e-8):
 
 
 class PowerSGDState(object):
-    __slots__ = ["process_group", "matrix_approximation_rank", "rng"]
-
-    def __init__(self, process_group, matrix_approximation_rank=1, random_seed=0):
+    __slots__ = [
+        "process_group",
+        "matrix_approximation_rank",
+        "use_error_feedback",
+        "rng",
+        "error_dict",
+    ]
+
+    def __init__(
+        self,
+        process_group,
+        matrix_approximation_rank=1,
+        use_error_feedback=True,
+        random_seed=0,
+    ):
         self.process_group = process_group
         self.matrix_approximation_rank = matrix_approximation_rank
+        # Error feedback is usually crucial for both for convergence and generalization,
+        # because PowerSGD is a biased compressor,
+        # i.e., compressing and decompressing a random gradient does not yield the original in expectation.
+        # This mechanism requires a temporary copy of the input gradients,
+        # so it increases the peak memory consumption.
+        # However, if the target matrices are known to be exactly low-ranked (instead of just low stable rank),
```

Comment:
When does this actually happen (target matrices are low-rank)?

Reply:
Good question! TBH, I don't see the answer in the referenced paper. According to PowerSGD paper, this should be uncommon.

I made the error feedback optional, so the user can choose a low-accuracy PowerSGD compression w/o such error feedback and reduce the peak memory usage.